src/darwin/Framework/CHIPTests/MTRBleTests.m

```diff
@@ -0,0 +1,192 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTRMockCB.h"
+#import "MTRTestCase.h"
+#import "MTRTestKeys.h"
+#import "MTRTestStorage.h"
+
+#import <Matter/Matter.h>
+#import <XCTest/XCTest.h>
+#import <stdatomic.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MTRBleTests : MTRTestCase
+@end
+
+@interface TestBrowserDelegate : NSObject <MTRCommissionableBrowserDelegate>
+@property (strong) void (^onDidFindCommissionableDevice)(MTRDeviceController *, MTRCommissionableBrowserResult *);
+@property (strong) void (^onDidRemoveCommissionableDevice)(MTRDeviceController *, MTRCommissionableBrowserResult *);
+@end
+
+@implementation TestBrowserDelegate
+
+- (void)controller:(nonnull MTRDeviceController *)controller didFindCommissionableDevice:(MTRCommissionableBrowserResult *)device
+{
+    __auto_type block = self.onDidFindCommissionableDevice;
+    if (block) {
+        block(controller, device);
+    }
+}
+
+- (void)controller:(nonnull MTRDeviceController *)controller didRemoveCommissionableDevice:(MTRCommissionableBrowserResult *)device
+{
+    __auto_type block = self.onDidRemoveCommissionableDevice;
+    if (block) {
+        block(controller, device);
+    }
+}
+
+@end
+
+MTRDeviceController * sController;
+
+@implementation MTRBleTests
+
+- (void)setUp
+{
+    [super setUp];
+
+    [self.class.mockCoreBluetooth reset];
+
+    sController = [MTRTestCase createControllerOnTestFabric];
+    XCTAssertNotNil(sController);
+}
+
+- (void)tearDown
+{
+    [sController shutdown];
+    sController = nil;
+    [[MTRDeviceControllerFactory sharedInstance] stopControllerFactory];
```

Comment:
Followup: MTRTestCase should perhaps handle this, since it handled the factory startup....  Need to think about it (which is why it's a followup).

Reply:
Yeah we would need some way to indicate if it's meant to be test scope or suite scope... maybe a class vs instance factory method, and then cleanup happens automatically with the correct scope.